source/nodejs/adaptivecards/src/serialization.ts

```diff
@@ -1086,10 +1086,23 @@ export abstract class SerializableObject {
         }
     }
 
-    protected getValue(prop: PropertyDefinition): any {
-        return this._propertyBag.hasOwnProperty(prop.getInternalName())
-            ? this._propertyBag[prop.getInternalName()]
-            : prop.defaultValue;
+    protected getValue(prop: PropertyDefinition, shouldOverrideProperty: boolean = false): any {
```

Comment:
minor question/nitpick -- is there any other scenario outside of when `GlobalSettings.enableFullJsonRoundTrip === true` you can envision where we would the `shouldOverrideProperty === true` ?

if so, I'm curious about it (this isn't supposed to be a "gotcha" question or me using the Socratic Method)... if there's no such scenario, then consider splitting out the `_rawProperties`-accessing version of this method into a new one (`getRawValue`, perhaps? could it be `private`?) and have `setValue` call it explicitly when `GlobalSettings.enableFullJsonRoundTrip` is `true`.

this helps eliminate confusion for future devs working in the area -- otherwise they may need to figure out what this `shouldOverrideProperty` parameter does in order to determine if they should supply it or not...

Reply:
I included `shouldOverrideProperty` because we don't want to access `_rawProperties` in every scenario even when `GlobalSettings.enableFullJsonRoundTrip === true`.

For instance, when the card is rendered, we call `getValue`. If we use the `_rawProperties` value, we get type errors (enums, templating expressions).

I only wanted `shouldOverrideProperty === GlobalSettings.enableFullJsonRoundTrip` when converting the card to JSON so that this piece of code in the designer gets a JSON with the expressions.

https://github.com/microsoft/AdaptiveCards/blob/4e987c5fbdd9725f9b769efc1c62af022a14e945/source/nodejs/adaptivecards-designer/src/card-designer-surface.ts#L292-L324